astropy/utils/misc.py

```diff
@@ -19,19 +19,21 @@
 import threading
 import re
 
-from itertools import zip_longest
 from contextlib import contextmanager
 from collections import defaultdict, OrderedDict
 
 from astropy.utils.decorators import deprecated
+# For backwards compatibility, import the shape-related stuff.
```

Comment:
What about these symbol definitions for back-compatibility?  Are there other uses of `ShapedLikeNDArray` outside astropy core?

Reply:
But at least I shouldn't leave commented-out code - I've now simply remove it.